src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -138,22 +138,25 @@ def check_bound(array, dim, index, operator):
             "".format(type(reference).__name__)
         assert reference.symbol is array.symbol, \
             "Expecting Reference symbol '{0}' to be the same as array " \
-            "symbol '{1}'.".format(reference.symbol.name, array.symbol.name)
+            "symbol '{1}'.".format(reference.symbol.name,
+                                   array.symbol.name)
         assert isinstance(literal, Literal), \
             "Expecting Literal but found '{0}'." \
             "".format(type(literal).__name__)
         assert literal.datatype == DataType.INTEGER, \
             "Expecting integer datatype but found '{0}'." \
             "".format(literal.datatype)
         assert literal.value == str(dim), \
-            "Expecting literal value '{0}' to be the same as the current " \
-            "array dimension '{1}'.".format(literal.value, str(dim))
+            "Expecting literal value '{0}' to be the same as the " \
+            "current array dimension '{1}'.".format(literal.value,
+                                                    str(dim))
     except AssertionError as excinfo:
```

Comment:
This code too assumes that `dim >= 1` and if the caller accidentally provides 0 then we will get an unhandled exception. Please document that `dim` must be >= 1 and make the code more robust, e.g. catch exceptions that aren't an `AssertionError` and handle them separately.

Reply:
Following my earlier comment on the use of 0 and 1 to indicate lower and upper bound, could we get rid of that argument entirely and use the type of the operator being supplied (i.e. LBOUND or UBOUND)?

Reply:
OK, I now robustly capture the arguments to ` _check_bound_is_full_extent` and use LBOUND and UBOUND to determine which index of the Range we are interested in.

Reply:
I'm much happier now :-)